### PR TITLE
New Event Dialog: Fix input validation

### PR DESCRIPTION
**Pull request by sevonj**
### Issues:
The  dialog was quite buggy.

- The _Add Event_ button would be enabled even if the date was invalid.
- The _Add Event_  button would not be disabled if the text box was cleared afterwards, unless the year box is touched.

### Fixes:
Now the button will light up if and only if all input is valid.
- All fields are properly validated for the _Add Event_ button to be enabled.
- All fields trigger rechecking of the _Add Event_ button's status.

### CODE REVIEW

File: src/Widgets/dialog.ui.vala [hunk 1]
```diff
@@ -26,25 +26,31 @@ namespace Countdown {
...
+        }
+
+        private DateTime get_datetime () {
+            return new DateTime.local (int.parse (event_date_year_entry.get_text ()),
+                                       int.parse (event_date_month_entry.get_text ()),
+                                       int.parse (event_date_day_entry.get_text ()),
+                                       0,
+                                       0,
+                                       0.0);
+        }
```
Comment (sevonj) [inline src/Widgets/dialog.ui.vala:43]: Extracted because of reuse.

File: src/Widgets/dialog.ui.vala [hunk 2]
```diff
@@ -26,25 +26,31 @@ namespace Countdown {
...
-                        new_button.sensitive = false;
-                    }
-                });
-            });
+
+            event_name_entry.notify["text"].connect (update_new_button);
+            event_date_day_entry.notify["text"].connect (update_new_button);
+            event_date_month_entry.notify["text"].connect (update_new_button);
+            event_date_year_entry.notify["text"].connect (update_new_button);
+        }
```
Comment (sevonj) [inline src/Widgets/dialog.ui.vala:34]: Every field triggers rechecking.

File: src/Widgets/dialog.ui.vala [hunk 3]
```diff
@@ -26,25 +26,31 @@ namespace Countdown {
...
+                                       0,
+                                       0,
+                                       0.0);
+        }
+
+        private void update_new_button () {
+            bool date_is_valid = get_datetime () != null;
+            bool name_is_valid = event_name_entry.get_text () != "";
+            new_button.sensitive = (date_is_valid && name_is_valid);
         }
```
Comment (sevonj) [inline src/Widgets/dialog.ui.vala:49]: The new validation function for `new_button`

**Comment by lainsce** [comment]
Thanks for your help~

**Pull request merged**